### ISSUE TITLE
Galley arbitrarily becoming required in 1.2.0 

### ISSUE THREAD

**Issue by Stono** [istio/istio#15029]
Hi, 
We're trying to upgrade one of our clusters from `1.1.9` to `1.2.0` and having a pretty horrendous time, mainly with the mixer components of the control plane crash looping.

It seems that as of `1.2.0`, galley is unofficially required, despite there being [nothing](https://istio.io/about/notes/1.2/) in the release notes about this.  There are several open issues regarding components failing or not working as expected without galley:

  - https://github.com/istio/istio/issues/14841
  - https://github.com/istio/istio/issues/14836
  - https://github.com/istio/istio/issues/12545

For context; we've not used galley _ever_.  We've never had time to properly test galley, and there's been no communication that it's a required component.  I've also opened a few other issues about Galley which make me question its production readiness:

  - https://github.com/istio/istio/issues/15091
  - https://github.com/istio/istio/issues/15088
  - https://github.com/istio/istio/issues/15098

Now that galley is seemingly being made unofficially required, we decided to try and deploy it, however the deployment failed.  We tried to roll back and galley actually stopped the rollback working.

```
❯ helm rollback istio 182
Error: admission webhook "pilot.validation.istio.io" denied the request: configuration is invalid: short names (non FQDN) are not allowed
```

As a result our only remaining option was to purge Istio and fresh deploy.   But wait, no, that doesn't work either because galley leaves behind an admission webhook so when you come to reinstall istio, it fails:

```
❯ kis get validatingwebhookconfigurations.admissionregistration.k8s.io
NAME           CREATED AT
istio-galley   2019-06-20T15:15:53Z
```

And then on top of that (not istios fault) but helm left lingering resources because of the failed deployment, so even after a purge I had to go and manually delete a bunch of galley bits from the cluster.

I'd really like to know what the deal is with Galley in 1.2.0 and onwards, eg:

  - If it's required, the option to disable it should be removed.
  - If it isn't required, non-galley routes should absolutely be tested.
  - If things become required (eg galley seems to be in 1.2.0) then that should be very clear in release notes.

**Comment by duderino** [comment]
@ozevren @mandarjog @costinm was this an unintentional dependency?  Do we need tests to ensure that components work properly without Galley?

**Comment by ozevren** [comment]
This was unintentional. I am diagnosing what has caused this. I think we should test the installation without Galley as well. I thought we had tests for this. If not, I'll look to add it.

**Comment by linsun** [comment]
We have a test for minimal istio profile which runs when galley is off but mixer is also off.

Sorry we didn't realize galley is absolutely required for 1.2 or required for mixer, so it is not intentionally left off from relnote.

**Comment by mandarjog** [comment]
I am looking at the problem now.

**Comment by Stono** [comment]
Thanks all.
As the default behaviour is to deploy galley, I am presuming it is stable?  

Basically I'm trying to best judge when to start migrating to Galley, and wondering if there are any outstanding issues (should I wait for 1.3+) or should I do it now in 1.2?

**Comment by sdake** [comment]
@Stono following up on your question related to horizontal scaling and the scaling listed as 1 in the manifest, @ozevren had indicated there is some minor improvement he intends to make to galley's sidecar handling to handle more scale.  I probably butchered what he said, so I'll let him add commentary there.

I use galley all the time (but not at very large scale) and galley works well).  I have launched 900 services with 900 deployments (across 3 physical nodes with lots of cpu and ram) and galley handles validation acceptably without failure.

**Comment by Stono** [comment]
I'm seeing numerous issues with galley testing:

 - https://github.com/istio/istio/issues/15088
 - https://github.com/istio/istio/issues/15091

I definitely cannot put this into a production environment.

Please can anyone confirm a `1.2.1` ETA removing the galley requirement?

**Comment by mandarjog** [comment]
@Stono the fix was just merged into the 1.2 branch.
The daily mixer build from tomorrow (June 25, release-1.2-20190624) will have the fix in it.
That is the *only* change between 1.2.0 Mixer. If you want you can test with this build in pre-prod.

1.2.1 will have a few other changes and will be released end of this week, of mid-next week.

**Comment by Stono** [comment]
That'd be great, thanks very much @mandarjog and team

**Comment by ozevren** [comment]
The original issue causing the Mixer to not work without Galley is fixed. There are other bugs tracking issues.

Do we still need this bug?

**Comment by fpesce** [comment]
> @Stono the fix was just merged into the 1.2 branch.
> The daily mixer build from tomorrow (June 25, release-1.2-20190624) will have the fix in it.
> That is the _only_ change between 1.2.0 Mixer. If you want you can test with this build in pre-prod.
> 
> 1.2.1 will have a few other changes and will be released end of this week, of mid-next week.

Or maybe a bit earlier for the impatient... https://gcsweb.istio.io/gcs/istio-prerelease/prerelease/1.2.1/
